src/main/java/com/endava/internship/collections/StudentSet.java

```diff
@@ -1,84 +1,339 @@
 package com.endava.internship.collections;
 
-import java.util.Collection;
-import java.util.Iterator;
-import java.util.Set;
+import java.time.LocalDate;
+import java.util.*;
+
+public class StudentSet implements Set<Student>, Iterable<Student> {
+    private Node root;
+    private int size;
+
+    StudentSet(){
+        root = null;
+        size = 0;
+    }
+    StudentSet(Collection<Student> collection) {
+        addAll(collection);
+    }
 
-public class StudentSet implements Set<Student> {
     @Override
     public int size() {
-        //TODO
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        //TODO
-        return false;
+        return size == 0;
     }
 
+    public Node getRoot(){
+        return root;
+    }
+    public Student getRootValue(){
```

Comment:
I think this is redundant, since you can do getRoot().value, also subject to NPE, since you directly call root which can be null